src/app/api/quote/route.ts

```diff
@@ -0,0 +1,141 @@
+      inputAmount,
+      outputToken: outputTokenAddress,
+      outputAmount,
+      slippage,
+    } = request
+
+    if (slippage < 0 || slippage > 1) {
+      return BadRequest('Bad Request')
+    }
+
+    if (!isAddress(inputTokenAddress) || !isAddress(outputTokenAddress)) {
+      return BadRequest('Bad Request')
+    }
+
+    if (!inputAmount && !outputAmount) {
+      return BadRequest(
+        'Either `inputAmount` or outputAmount` needs to be set.',
+      )
+    }
+
+    if (inputAmount && outputAmount) {
+      return BadRequest('You can only set `inputAmount` or outputAmount`.')
+    }
+
+    const inputToken = getQuoteToken(inputTokenAddress, chainId)
+    const outputToken = getQuoteToken(outputTokenAddress, chainId)
+
+    if (
+      !inputToken ||
+      !outputToken ||
+      (inputToken.isIndex && outputToken.isIndex)
+    ) {
+      return BadRequest('Bad Request')
+    }
+
+    const rpcUrl =
+      getAlchemyBaseUrl(chainId) + process.env.NEXT_PUBLIC_ALCHEMY_ID
+    const zeroexSwapQuoteProvider =
+      getConfiguredZeroExSwapQuoteProvider(chainId)
+    const quoteProvider = new FlashMintQuoteProvider(
+      rpcUrl,
+      zeroexSwapQuoteProvider,
+    )
+
+    const quote = await quoteProvider.getQuote({
+      isMinting: outputToken.isIndex,
+      inputToken: inputToken.quoteToken,
+      outputToken: outputToken.quoteToken,
+      indexTokenAmount: BigNumber.from(parseEther('1').toString()),
+      slippage,
+    })
+
+    if (!quote) {
+      return NextResponse.json({ message: 'No quote found.' }, { status: 404 })
+    }
+
+    return NextResponse.json({
+      type: QuoteType.flashmint,
+      chainId: quote.chainId,
+      contract: quote.contract,
+      contractType: quote.contractType,
+      takerAddress: account,
+      isMinting: quote.isMinting,
+      inputToken: quote.inputToken,
+      outputToken: quote.outputToken,
+      indexTokenAmount: quote.indexTokenAmount.toString(),
+      inputAmount: quote.inputAmount.toString(),
+      outputAmount: quote.outputAmount.toString(),
+      slippage: quote.slippage,
+      transaction: quote.tx,
+    })
+  } catch (error) {
+    console.error(error)
+    return NextResponse.json(error, { status: 500 })
+  }
+}
+
+function BadRequest(errorMessage: string) {
+  return NextResponse.json({ message: errorMessage }, { status: 400 })
+}
+
+function getQuoteToken(
+  token: Address,
+  chainId: number,
+): { quoteToken: QuoteToken; isIndex: boolean } | null {
+  if (isAddressEqual(token, EthAddress)) {
+    return {
+      quoteToken: {
+        symbol: 'ETH',
+        decimals: 18,
+        address: EthAddress,
+      },
+      isIndex: false,
+    }
+  }
+  const listedToken = getTokenByChainAndAddress(chainId, token)
+  if (!listedToken) return null
+  return {
+    quoteToken: {
+      symbol: listedToken.symbol,
+      decimals: listedToken.decimals,
+      address: listedToken.address,
+    },
+    isIndex: isIndexToken(listedToken),
```

Comment:
instead of isIndexToken, you might want to use isProductToken here, index token includes the INDEX (symbol) token so all of our tokens.